junit-platform-commons/src/main/java/org/junit/platform/commons/support/SearchOption.java

```diff
@@ -37,7 +39,11 @@ public enum SearchOption {
 	 * Search the inheritance hierarchy as with the {@link #DEFAULT} search option
 	 * but also search the {@linkplain Class#getEnclosingClass() enclosing class}
 	 * hierarchy for <em>inner classes</em> (i.e., a non-static member classes).
+	 *
+	 * @deprecated because it's preferable to inspect the runtime enclosing
+	 * types of a class rather than where they are declared.
```

Comment:
Although I certainly agree with that sentiment, developers and libraries that use `AnnotationSupport` often will not have access to the runtime enclosing types. Typically only JUnit internals have access to those.

So, I am a bit reluctant to deprecate this feature.

However, if we deprecate it and nobody complains, then I suppose it's fine. 😇

Reply:
> Although I certainly agree with that sentiment, developers and libraries that use `AnnotationSupport` often will not have access to the runtime enclosing types. Typically only JUnit internals have access to those.

Jupiter extension authors can access them via `ExtensionContext.getRequiredTestClass()`, recursively, or get them from `ExtensionContext.getRequiredTestInstances()` by calling `Object.getClass()` on each instance.

I checked the original issue and didn't find an actual user request. Therefore, since this was the only place we used it, I deprecated the method. We can always undeprecate it if anyone complains. WDYT?

Reply:
I also considered the fact that Jupiter extension authors have a way to access that information, so I was thinking more about the general purpose nature of the utilities in `AnnotationSupport` for other testing frameworks, engines, etc.

> I checked the original issue and didn't find an actual user request. Therefore, since this was the only place we used it, I deprecated the method. We can always undeprecate it if anyone complains. WDYT?

Yep, as I mentioned before.... if nobody complains, it's fine. 👍